web/source/settings/views/admin/debug/apurl/index.tsx

```
@@ -53,6 +53,9 @@ export default function ApURL() {
 				<TextInput
 					field={urlField}
 					label="URL"
+					type="url"
+					pattern="(http|https):\/\/.+"
+					placeholder="https://example.org/users/someone"
```

Do we want to set a `minlength` on these too?

---

Do we need to set the `'u'` flag here to allow for unicode?

---

Ah good question, I'll check...

---

Seems to work OK with unicode characters already :) `https;//üxample.org` is valid.

Instead of a min-length I guess what we could do is try parsing the field value into a URL, and fail validation if it throws, similar to what we do for RE2 regexes now in the header permissions form. What do you think?

---

Oof OK I tried it with parsing a URL but that's somehow even clunkier and doesn't always return invalid even for really wonky URLs, so I think probably just keeping the regex in this PR is good enough for now..

---

The only reason I suggest `minlength` was to ensure we got at least `http://` without even bothering with the pattern match since anything less than 7 characters won't match anyway. It's probably fine as is though.

---

Aha okay! Yeah this is probably fine then, and a bit more explicit about what we're looking for :)